File: src/main/java/hzt/aoc/day04/PassportProcessingPart2.java
package hzt.aoc.day04;

import hzt.aoc.day04.model.Passport;
import hzt.collections.ListX;

public class PassportProcessingPart2 extends Day04Challenge {

    public PassportProcessingPart2() {
        super("part 2", "Find the number of valid passports. ");
    }

    @Override
    protected long calculateResult(ListX<Passport> passports) {
        return passports.count(Passport::fieldsMeetCriteria);
    }

}

File: src/main/kotlin/hzt/aoc/day04/PassportProcessingPart2.kt
package hzt.aoc.day04

import hzt.aoc.day04.model.Passport

class PassportProcessingPart2 : Day04Challenge("part 2", "Find the number of valid passports. ") {

    override fun calculateResult(passports: List<Passport>): Long =
        passports.count(Passport::fieldsMeetCriteria).toLong()
}
